### ISSUE TITLE
Evaluation of the robustness

### ISSUE THREAD

**Issue by Barath19** [ika-rwth-aachen/MultiCorrupt#5]
Hello Till,

I have successfully created the corruption dataset for just spatialmisalignment and for all the three severity level. However. when I run the evaluation for BEVFusion (mmdetection3d) I see no difference in the result compared the clean version. I assume the template evaluation script provided does a symbolic link to the root nuscenes dataset and the corruption dataset.  Is it possible to run the evaluation just for spatialmisalignment. In that case the output folder just contains the  LIDAR_TOP folder, how does the symbolic link work in this case.

TLDR; how can we use the generated corruption data to run the evaluation. Apparently the results for all the severity is the same (i.e. NDS  0.7113).

Best,
Barath

**Comment by TillBeemelmanns** [comment]
The template evaluation script does a symbolic link from

```
multicorrupt_root="/workspace/multicorrupt/"
```
to
```
nuscenes_data_dir="/workspace/data/nuscenes"
```
i.e.

```
ln -s "$multicorrupt_root/$corruption/$severity" "$nuscenes_data_dir"
```

You would need to double check the pathes and check in the terminal if the symbolic link is actually working. 

Also make sure, that directories `maps` and `v1.0-trainval` is copied into the respective directories, like this

![image](https://github.com/ika-rwth-aachen/MultiCorrupt/assets/5880684/9d829e96-c5ba-4c8f-8eb3-7af2c9a662d9)

**Comment by TillBeemelmanns** [comment]
The corrupted dataset of severity 1 needs to have the same structure as the original nuScenes dataset. Just the corrupted files need to be replaced. So apparently `CAM_FRONT` and the other camera dirs are missing in your setup. The structure needs to like the following. 

```
1
|-- maps
|   |-- 36092f0b03a857c6a3403e25b4b7aab3.png
|   |-- 37819e65e09e5547b8a3ceaefba56bb2.png
|   |-- 53992ee3023e5494b90c316c183be829.png
|   `-- 93406b464a165eaba6d9de76ca09f5da.png
|-- samples
|   |-- CAM_BACK
|   |-- CAM_BACK_LEFT
|   |-- CAM_BACK_RIGHT
|   |-- CAM_FRONT
|   |-- CAM_FRONT_LEFT
|   |-- CAM_FRONT_RIGHT
|   `-- LIDAR_TOP
|-- sweeps
|   `-- LIDAR_TOP
`-- v1.0-trainval
    |-- attribute.json
    .
    .
    .
```
You can simply copy and paste the missing directories from the original dataset there, or create a soft link. Then it should work. 

I will check the code and see how I can automate this. But I will probably have time for that next month, earliest.

**Comment by Barath19** [comment]
Oh cool...sorted!! I guess this was missing in my understanding of the evaluation script. Thanks alot for feedbacks. I am working on the same. If you can update the contribution guidelines I can create a pull request when completed.
